src/CMakeLists.txt

```diff
@@ -27,3 +28,166 @@ add_library(bitcoin_consensus OBJECT EXCLUDE_FROM_ALL
   util/strencodings.cpp
 )
 target_link_libraries(bitcoin_consensus PRIVATE secp256k1)
+
+
+# Home for common functionality shared by different executables and libraries.
+# Similar to `bitcoin_util` library, but higher-level.
+add_library(bitcoin_common STATIC EXCLUDE_FROM_ALL
+  base58.cpp
+  bech32.cpp
+  chainparams.cpp
+  coins.cpp
+  common/bloom.cpp
+  common/interfaces.cpp
+  common/run_command.cpp
+  $<$<TARGET_EXISTS:PkgConfig::libevent>:common/url.cpp>
+  compressor.cpp
+  core_read.cpp
+  core_write.cpp
+  deploymentinfo.cpp
+  external_signer.cpp
+  init/common.cpp
+  key.cpp
+  key_io.cpp
+  merkleblock.cpp
+  net_types.cpp
+  netaddress.cpp
+  netbase.cpp
+  net_permissions.cpp
+  outputtype.cpp
+  policy/feerate.cpp
+  policy/policy.cpp
+  protocol.cpp
+  psbt.cpp
+  rpc/rawtransaction_util.cpp
+  rpc/request.cpp
+  rpc/external_signer.cpp
+  rpc/util.cpp
+  scheduler.cpp
+  script/descriptor.cpp
+  script/miniscript.cpp
+  script/sign.cpp
+  script/signingprovider.cpp
+  script/standard.cpp
+  warnings.cpp
+)
+target_link_libraries(bitcoin_common
+  PRIVATE
+    bitcoin_consensus
+    bitcoin_util
+    univalue
+    secp256k1
+    Boost::boost
+    Threads::Threads
```

Comment:
Currently the treads flags / libs are included when linking the final binary. I'm guessing it is out of necessity that they are linked into the libraries now. Is there a performance/size downside to statically linking them to both the common and util library?

Reply:
eh? Catching up, but why is Boost even in here. We don't link any Boost libs, and shouldn't be linking any ever

Reply:
> eh? Catching up, but why is Boost even in here. We don't link any Boost libs, and shouldn't be linking any ever

Tee [`target_link_libraries()`](https://cmake.org/cmake/help/latest/command/target_link_libraries.html) command also propagates usage requirements from the listed targets.

For the [`Boost::boost`](https://cmake.org/cmake/help/latest/module/FindBoost.html#imported-targets) target, which is a headers only dependency, only an include directory is a such usage requirement.

Reply:
I think an alternative here would be maintaining boost header propagation ourselves, by requiring that clients of libs using boost headers also opt-in to boost headers via `target_include_directories`.

I don't have a problem with keeping this as-is though, as well as it's well-behaved and thoroughly commented.

Reply:
> Currently the threads flags / libs are included when linking the final binary. I'm guessing it is out of necessity that they are linked into the libraries now.

The idea is to provide direct dependencies for each build target/artifact. Such an approach make the usage those artifacts much cleaner. Consider `target_link_libraries(bitcoind bitcoin_node)`.

> Is there a performance/size downside to statically linking them to both the common and util library?

Tbh, I didn't made such a comparison.